### DISCUSSION TITLE
Prisma Migrate - Missing Foreign Keys

### DISCUSSION THREAD

**Discussion by webnoob** [prisma/prisma#4237] [Q&A] [answered]
Hi,

A snippet of my schema is:

```prisma
model PenName {
  id Int  @default(autoincrement()) @id
  createdDate DateTime @default(now())
  name String

  user User @relation(fields: [userId], references: [id])
  userId Int

  @@unique([id, userId], name: "penName_id_userId")
}

model User {
  id Int  @default(autoincrement()) @id
  createdDate DateTime @default(now())
  emailAddress String @unique
}
```

I would expect to see some foreign keys assigned on my PenName table but I don't. Is this a limitate of using Prisma migrate at the moment? Do I need to use introspection instead?

Note: When I say I can't see the FK's, I can't see them in MySQL Workbench OR via this query:

```sql
SELECT 
  TABLE_NAME,COLUMN_NAME,CONSTRAINT_NAME, REFERENCED_TABLE_NAME,REFERENCED_COLUMN_NAME
FROM
  INFORMATION_SCHEMA.KEY_COLUMN_USAGE
WHERE
  REFERENCED_TABLE_SCHEMA = 'dev' AND
  REFERENCED_TABLE_NAME = 'penname';
```

**Comment by ryands17** [comment, marked as answer]
@webnoob The foreign key is created when you run the following commands on the above schema:

```
prisma migrate save --experimental
prisma migrate up --experimental
```

Then a `migrations` folder will be creatd locally and you will be able to the SQL generated in the `README.md` which will look something like this:

```sql
CREATE TABLE `PenName` (
    `id` INT NOT NULL AUTO_INCREMENT,
    `createdDate` DATETIME(3) NOT NULL DEFAULT CURRENT_TIMESTAMP(3),
    `name` VARCHAR(191) NOT NULL,
    `userId` INT NOT NULL,
UNIQUE INDEX `penName_id_userId`(`id`,
`userId`),

    PRIMARY KEY (`id`)
) DEFAULT CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci

CREATE TABLE `User` (
    `id` INT NOT NULL AUTO_INCREMENT,
    `createdDate` DATETIME(3) NOT NULL DEFAULT CURRENT_TIMESTAMP(3),
    `emailAddress` VARCHAR(191) NOT NULL,
UNIQUE INDEX `User.emailAddress_unique`(`emailAddress`),

    PRIMARY KEY (`id`)
) DEFAULT CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci

ALTER TABLE `PenName` ADD FOREIGN KEY (`userId`) REFERENCES `User`(`id`) ON DELETE CASCADE ON UPDATE CASCADE
```

So a foreign key is created on Migrate.

**Reply by webnoob** [reply]
That's what I thought - the thing is, it isn't ...

![image](https://user-images.githubusercontent.com/10404747/98937507-63c31100-24de-11eb-8d23-11576580ac3d.png)

**Reply by webnoob** [reply]
Just checking the readme.md now ...

**Reply by webnoob** [reply]
Hmmmm, Ok - the line is there: ```ALTER TABLE `PenName` ADD FOREIGN KEY (`userId`) REFERENCES `dev`.`User`(`id`) ON DELETE CASCADE ON UPDATE CASCADE``` but it's not created in the DB..

I wonder if this is down to the current windows bug where the migration fails because of case sensitivity and thus doesn't get as far as creating the FK's.

**Reply by ryands17** [reply]
I tried the same schema and here's the Adminer screenshot. Also I'm on version 2.11.0

![image](https://user-images.githubusercontent.com/19697099/98937658-badfda80-250c-11eb-90bf-eb2d04b5677a.png)

**Reply by ryands17** [reply]
> I wonder if this is down to the current windows bug where the migration fails because of case sensitivity and thus doesn't get as far as creating the FK's.

I would suggest creating an issue [here](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=bug_report.md&title=) then. I'm not on Windows so won't be able to reproduce.

**Reply by webnoob** [reply]
There is already an issue outstanding for this: https://github.com/prisma/prisma-engines/issues/950

I'll try a work around which was suggested and see if, when the migration completes, the FK's exist. I'm sure they will as I think this will be related. Just wanted to make sure migrate was supposed to work properly with this 👍

**Reply by webnoob** [reply]
Excellent - after applying the workaround here: https://github.com/prisma/migrate/issues/339#issuecomment-713226205

I wasn't too fussed with the workaround as I didn't think there was any side effects. Happy to have this working for now.

It works: 
![image](https://user-images.githubusercontent.com/10404747/98938542-f6b07b00-24df-11eb-8ff5-e57699249d81.png)

**Reply by webnoob** [reply]
Sorry, how rude of me ...

Many thanks for your speedy and helpful assistance.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
